x-pack/plugins/cases/public/components/custom_fields/custom_fields_list/index.tsx

```diff
@@ -16,25 +16,46 @@ import {
   EuiSpacer,
   EuiText,
   EuiIcon,
+  EuiButtonIcon,
+  useEuiTheme,
 } from '@elastic/eui';
 
 import { css } from '@emotion/react';
 import type { CustomFieldTypes, CustomFieldsConfiguration } from '../../../../common/types/domain';
 import { builderMap } from '../builder';
+import { DeleteConfirmationModal } from '../delete_confirmation_modal';
 
 export interface Props {
   customFields: CustomFieldsConfiguration;
+  onDeleteCustomField: (key: string) => void;
 }
 
 const CustomFieldsListComponent: React.FC<Props> = (props) => {
-  const { customFields } = props;
+  const { customFields, onDeleteCustomField } = props;
+  const { euiTheme } = useEuiTheme();
+  const [showDeletionModal, setShowDeletionModal] = useState<{
```

Comment:
nit: if you want to simplify a bit the code you can do `const [selectedItem, setSelectedItem] = useState<CustomFieldsConfiguration[number] | null>(null)`. Then you can do `const showModal = Boolean(selectedItem)`. Also,

```
onClick={() =>
  setSelectedItem(customFields)
}
```